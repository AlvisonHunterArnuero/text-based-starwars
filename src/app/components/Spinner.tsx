import { Box, Typography } from '@mui/material';
import CircularProgress from '@mui/material/CircularProgress';
import React from 'react';

export const Spinner = ({ title }: { title: string }) => {
  return (
    <Box
      sx={{
        display: 'flex',
        minHeight: '80vh',
        alignItems: 'center',
        justifyContent: 'justify-between',
        gap: 3,
      }}
    >
      <CircularProgress />
      <Typography variant="h4">{title}</Typography>
    </Box>
  );
};
