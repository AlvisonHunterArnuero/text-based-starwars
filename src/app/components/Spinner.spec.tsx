import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, it, expect } from 'vitest';

import '@testing-library/jest-dom/vitest';
import { Spinner } from './Spinner';

describe('Spinner Component', () => {
  const testTitle = 'Loading...';

  it('renders without crashing', () => {
    render(<Spinner title={testTitle} />);
    expect(screen.getByRole('progressbar')).toBeInTheDocument();
  });

  it('displays the correct title prop', () => {
    render(<Spinner title={testTitle} />);
    expect(screen.getByText(testTitle)).toBeInTheDocument();
  });

  it('contains a Material-UI CircularProgress', () => {
    render(<Spinner title={testTitle} />);
    const progressBar = screen.getByRole('progressbar');
    expect(progressBar).toBeInTheDocument();
    expect(progressBar).toHaveClass('MuiCircularProgress-root');
  });

  it('applies correct flexbox styles', () => {
    render(<Spinner title={testTitle} />);
    const box = screen.getByTestId('spinner-box');
    expect(box).toHaveStyle('display: flex');
  });

  it('matches snapshot', () => {
    const { container } = render(<Spinner title={testTitle} />);
    expect(container).toMatchSnapshot();
  });
});
