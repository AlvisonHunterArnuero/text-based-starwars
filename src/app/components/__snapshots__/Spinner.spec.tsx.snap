// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner Component > matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-gulyk2"
    data-testid="spinner-box"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-19t5dcl-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
    <h4
      class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
    >
      Loading...
    </h4>
  </div>
</div>
`;
